backend/src/main/java/com/woowacourse/friendogly/club/domain/Club.java

```diff
@@ -158,11 +162,23 @@ public void addClubPet(List<Pet> pets) {
     }
 
     private void validateParticipatePet(Pet pet) {
-        if (!allowedGenders.contains(pet.getGender()) || !allowedSizes.contains(pet.getSizeType())) {
+        if (canNotJoin(pet)) {
             throw new FriendoglyException("모임에 데려갈 수 없는 강아지가 있습니다.");
         }
     }
 
+    private boolean canNotJoin(Pet pet) {
+        return !allowedGenders.contains(pet.getGender()) || !allowedSizes.contains(pet.getSizeType());
+    }
```

Comment:
```suggestion
        if (!canJoin(pet)) {
            throw new FriendoglyException("모임에 데려갈 수 없는 강아지가 있습니다.");
        }
    }

    private boolean canJoin(Pet pet) {
        return allowedGenders.contains(pet.getGender()) && allowedSizes.contains(pet.getSizeType());
    }
```

not 절이 많아서 읽기 어려울 수 있겠네요
not이 2개인 것 보다는 하나인 게 나을 것 같은데, 이렇게 바꾸는 건 어떨까요?

Reply:
이건 그냥 제안인데
자연어에 가까운 메서드 네이밍을 해보자면
`club.canBeJoinedBy(pet)` 이런 이름도 괜찮을 것 같아요

Reply:
canJoinWith로 변경했습니다.